### PR TITLE
Incantations to enforce Python checks

### PR DESCRIPTION
**Pull request by pllim**
This is for cookiecutter branch. Same as #301 

Companion of astropy/astropy#7316

### CODE REVIEW

**Comment by Cadair** [comment]
What @bsipocz said :laughing:

**Comment by pllim** [comment]
What if people don't want to check for Python version? What do they do in this set up?

**Comment by Cadair** [comment]
you could use a template to not include the whole section if the user puts "False" or something as the variable, i.e.:
https://github.com/astropy/package-template/blob/cookiecutter/%7B%7B%20cookiecutter.package_name%20%7D%7D/docs/conf.py#L114

**Comment by pllim** [comment]
I *think* I implemented your requests. Please double-check. Thank you!

**Comment by Cadair** [comment]
looks great @pllim, I think it would be good to test this, but it would probably have to be special cased in the travis yml. :thinking:

File: .travis.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ env:
       - EXTRA_CONTEXT='include_example_cython_code=y'
       - EXTRA_CONTEXT="package_name=AstropyProject" FOLDERNAME='AstropyProject'
       - EXTRA_CONTEXT='_parent_project=sunpy'
+      - EXTRA_CONTEXT='__minimum_python_version__=3.5'
```
Comment (pllim) [inline .travis.yml]: @Cadair , you mean like this? I can't say I am familiar with template testing.

Comment (Cadair) [reply .travis.yml]: yes I think so.

**Comment by pllim** [comment]
I added a test and it passed. FYI.

**Comment by bsipocz** [comment]
Thanks @pllim!

**Pull request merged**